Utility/Utility.Static/Logger.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <chrono>
+#include <ctime>
+#include <condition_variable>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <mutex>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <queue>
+
+class Logger final
+{
+public:
+    using BlockWrapper = std::pair<std::string, std::string>;
+
+public:
+    // Valid log levels
+    // ERROR - Task cannot continue and component cannot function
+    // WARNING - Potential error or impending error
+    // INFO - General information outlining overall task progress
+    // DEBUG - General information detailing debug progress
+    enum class LogLevel
+    {
+        ERROR,
+        WARNING,
+        INFO,
+        DEBUG
+    };
+
+    // Place of output of logged information
+    enum class LogOutput
+    {
+        CONSOLE,
+        FILE,
+        EVERYWHERE
+    };
+
+    // Stringify the log level to its name
+    // @param required LogLevel
+    inline std::string stringify(const LogLevel level);
+
+public:
+    Logger();
```

Comment:
Why can we create the objects outside this class, it it's a singleton? Maybe it would be better to move constructor in private section?